Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift

```diff
@@ -232,7 +232,7 @@ open class ChatChannelListViewModel: ObservableObject, ChatChannelListController
         channel.membership != nil
     }
 
-    func checkTabBarAppearance() {
+    public func checkTabBarAppearance() {
```

Comment:
should remain internal.

Reply:
Hello @martinmitrevski!

Thank for taking the time to review this.

As I wanted to customize `ChatChannelInfoView`, I needed to work backwards and replace it with `CustomChatChannelInfoView` (my version). For this view to be recognized, I had to work backwards all the way to the root invocation:

```swift
struct MessagingScreen: View {
    @Injected(\.appModel) private var appModel: AppModel

    var body: some View {
        CustomChannelListView(
            viewFactory: CustomFactory.shared,
            title: "Channels"
        )
        .foregroundStyle(appModel.userAccentUIColor.toColor)
    }
}
```

As I think about this now, would have been better to just fork `stream-chat-swiftui` and simply customize `CustomChatChannelInfoView`? Then it'd be a matter of keeping up with the Stream's parent repo. Honestly, I don't know why I didn't think of this before.

Any suggestions how to best approach this? 🙏🏻

Reply:
Hey @tciuro,

You can do that as well - you just need to make sure that you update regularly with our `main` branch, so you get the latest updates and features.

If it's just about the `CustomChatChannelInfoView`, I think it's better to use the official releases and re-use stuff as much as possible from the exposed components. For the internal ones, you can copy them in your project, which is still lower maintenance then keeping up a fork up to date.

In any case, it's up to you.